File: CS-Examples/07_Conversion/ToCSVWithDoubleQuotes.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace ToCSVWithDoubleQuotes
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of the Workbook class.
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ToCSV.xlsx")

            ' Convert the workbook to a CSV file.
            ' When the last parameter is set to true, double quotes will be added around each field. The default value is false.
            workbook.SaveToFile("ToCSVAddQuotation.csv", ",", True)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'view the document
            ExcelDocViewer("ToCSVAddQuotation.csv")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/07_Conversion/ToCSVWithDoubleQuotes.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace ToCSVWithDoubleQuotes
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load a excel document
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ToCSV.xlsx");

            // Convert to CSV file,
            // When the last parameter is set to true, there are double quotes. The default parameter is flase
            workbook.SaveToFile("ToCSVAddQuotation.csv", ",", true);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launsh the document
            ExcelDocViewer("ToCSVAddQuotation.csv");
        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
